### ISSUE TITLE
[Feature Request] Osm Editing: Allow converting (address) nodes to POI

### ISSUE THREAD

**Issue by matthijskooijman** [osmandapp/Osmand#5853]
In the Netherlands, address nodes are automatically imported into Osm from the official government database. This means that for each official house number, there is a pre-existing Osm node on the map. When mapping POIs, it is recommended to add additional tags onto these address nodes, rather than creating additional POI nodes (at least for the common case where a single house number contains a single shop or other POI).

Currently (3.0.4), OsmAnd allows selecting these house number nodes (which are shown as type "Location"), but when I want to add POI information I can select "Create POI", which creates a new, blank node, without any address information.

The easiest way to fix this would be let a "Location" node (e.g. a node with an address, I presume) be flagged as an existing POI when determining whether to show the "Create POI" or "Modify POI" action. Then you can just select "Modify POI" to modify the existing node, rather than having to create a new one.

**Comment by vshcherb** [comment]
I'm not sure it is a good idea or not cause POI like restaurants probably should not reuse housenumber points

**Comment by matthijskooijman** [comment]
> I'm not sure it is a good idea or not cause POI like restaurants probably should not reuse housenumber points

It seems that there are two approaches: Tag POIs next to the address (and deduce the POI address from proximity), or tag POIs on the same node as the address. There does not seem to be a clear consensus about this, e.g. from https://wiki.openstreetmap.org/wiki/Addresses:

> Usually address informations doesn't need to be duplicated, i.e. there should be only one addr:housenumber=* per street and house number (which would typically be placed on the building). Software can usually link adresses to other features by geographical proximity, and duplicated addresses are often flagged as errors by quality assurance tools. However, there is still some debate on that point (see for example [Address information in POI *and* building?](https://help.openstreetmap.org/questions/48735/address-information-in-poi-and-building) on help.openstreetmap.org). Also, the community in some countries has established their own rules. 

IMHO, tagging a POI together with the address can be beneficial, when multiple addresses are nearby (e.g. in a multi-storey building), then th POI on the address node makes the relation more explicit.

Regardless, this is also the common and recommended approach in the Netherlands, e.g. from https://wiki.openstreetmap.org/wiki/Welkom_op_OSM:

> Adressen worden ook automatisch op de panden geplaatst. Als je iets wilt toevoegen aan een bepaald adres (bv. dat er een schoenenwinkel zit), dan kun je dat het beste doen door aan de adresnode van het bedrijf, die gegevens (dus winkeltype, naam, website, enz) toe voegen. Je moet ze liever niet op het gebouw zelf zetten.

Which translates to:
> Addresses are also added to buildings automatically. If you want to add something to a certain address (e.g. that it houses a shoe store), then you can best add extra data (type of store, name, website, etc) to the addressnode of the company. You should rather not add them to the building itself.

In a way, the Netherlands might be slightly different here, since address nodes have been imported from a government registry, so pretty much all addresses are precreated as separate nodes and thus addresses are never tagged on buildings, which makes them more suited to reuse as a POI node (I think).

**Comment by vincentvd1** [comment]
I agree with Mathijs that such functionality is needed in OsmAnd. We see changesets where POI's are added via OsmAnd and the POI only contain the amenity or name tag. This while in the Netherlands, all address are present. Also, in other countries though are POI added on address nodes. 

Having the address on a POI increases the information quality and requires the need for another mapper to merge the nodes. 

It is also confusing that you can select an address in OsmAnd but adding a POI adds a new, empty POI
